hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMemoryConfig.java

```diff
@@ -57,13 +53,17 @@ public class HoodieMemoryConfig extends HoodieConfig {
   public static final ConfigProperty<String> MAX_MEMORY_FRACTION_FOR_COMPACTION = ConfigProperty
       .key("hoodie.memory.compaction.fraction")
       .defaultValue(String.valueOf(0.6))
+      .withAlternatives("compaction.memory.fraction")
```

Comment:
Where does the alternative come from, should it start with `hoodie.` prefix?

Reply:
Ohh, I see it's hard coded in the code before, should we keep it in release 1.0? Maybe we can just drop it directly to simplify the code.

Reply:
`compaction.memory.fraction` is used in `HoodieRealtimeConfig` before, in multiple places.  Since it's not hard to make it backward compatible, I still keep it so that the user does not have to change the config name.

Reply:
I was thinking because it is release 1.0, it is a good chance we can clean these legacy codes.

Reply:
Got it.  I think it makes sense.  These two config names are removed and consolidated:
- `compaction.memory.fraction` -> `hoodie.memory.compaction.fraction`
- `compaction.lazy.block.read.enabled` -> `hoodie.compaction.lazy.block.read`